module/VuFind/src/VuFind/Record/Router.php

```diff
@@ -85,14 +85,18 @@ public function getActionRouteDetails($driver, $action)
      * @param \VuFind\RecordDriver\AbstractBase|string $driver Record driver
      * representing record to link to, or source|id pipe-delimited string
      * @param string                                   $tab    Action to access
+     * @param array                                    $query  Optional query params
      *
      * @return array
      */
-    public function getTabRouteDetails($driver, $tab = null)
+    public function getTabRouteDetails($driver, $tab = null, $query = [])
     {
         $route = $this->getRouteDetails(
             $driver, '', empty($tab) ? [] : ['tab' => $tab]
         );
+        if ($query) {
```

Comment:
Do we need the `if ($query) {` check? I believe that the default empty array will evaluate to false and cause the query to go uninitialized. Is that intended?

Reply:
Will remove.

Reply:
Actually, that would cause an empty element to be added to the route definition and tests to fail. I think it's cleaner to only add the options element if it's needed.

Reply:
That makes sense. Might be worth adding a comment or switching to `if (!empty($query)) {` to be more explicit. Otherwise, I might look at this again in the future, forget this conversation, and mistake it for a bug.